cmd/config/config.go

```diff
@@ -119,68 +119,29 @@ func run(service config.Service) func(*cobra.Command, []string) error {
 
 			fmt.Fprintf(cmd.OutOrStdout(), output+"\n")
 		case viper.GetBool(SetFlag):
-			// flag needs two arguments: a key and value
-			if len(args)%2 != 0 {
-				return errs.NewError("flag needs an argument: --set")
-			}
-
-			for i := 0; i < len(args)-1; i += 2 {
-				_, ok := cliflags.AllFlagsHelp()[args[i]]
-				if !ok {
-					return newErr(args[i])
-				}
-			}
-
-			rawConfig, v, err := getRawConfig()
+			conf, err := config.New(viper.ConfigFileUsed(), os.ReadFile)
```

Comment:
There's a new `Config` type that encapsulates updating the config file. The "unset" action doesn't use it yet, so there is still duplicate code being used.